src/webnn_native/openvino/GraphIE.cpp

```diff
@@ -521,6 +521,101 @@ namespace webnn_native { namespace ie {
+            n++;
+        } else {
+            std::vector<size_t> biasShape{numDirections, 3 * hiddenSize};
+            std::vector<float> biasData(numDirections * 3 * hiddenSize, 0);
+            biasNode = AddConstantWithGraph<float>(precision_e::FP32, biasShape, biasData);
+        }
+        ngraph_node_t* recurrentBiasNode = nullptr;
+        if (options->recurrentBias != nullptr) {
+            recurrentBiasNode = const_cast<ngraph_node_t*>(mGraphNodeMap[inputs[n].Get()]);
+            n++;
+        } else {
+            std::vector<size_t> recurrentBiasShape{numDirections, 3 * hiddenSize};
+            std::vector<float> recurrentBiasData(numDirections * 3 * hiddenSize, 0);
+            recurrentBiasNode = AddConstantWithGraph<float>(precision_e::FP32, recurrentBiasShape,
+                                                            recurrentBiasData);
+        }
+        ngraph_node_t* initialHiddenStateNode = nullptr;
+        if (options->initialHiddenState != nullptr) {
+            initialHiddenStateNode = const_cast<ngraph_node_t*>(mGraphNodeMap[inputs[n].Get()]);
+            n++;
+        } else {
+            std::vector<size_t> initialHiddenStateShape{numDirections, batchSize, hiddenSize};
+            std::vector<float> initialHiddenStateData(numDirections * batchSize * hiddenSize, 0);
+            initialHiddenStateNode = AddConstantWithGraph<float>(
+                precision_e::FP32, initialHiddenStateShape, initialHiddenStateData);
+        }
+        ngraph_node_t* initialHiddenStateTransposeNode = nullptr;
+        status =
+            ngraph_transpose(initialHiddenStateNode, orderNode, &initialHiddenStateTransposeNode);
+        if (status != IEStatusCode::OK) {
+            dawn::ErrorLog() << "Failed to transpose gru initialHiddenState layout";
+        }
+        auto direction = static_cast<ngraph_recurrent_sequence_direction>(options->direction);
+        // TODO: layout
+        /**
+         * If resetAfter is set to true, then the bias shape will  be set to
+         * [num_directions, 4 * hidden_size] (not support)
+         */
+        bool linear_before_reset = options->resetAfter == true ? false : false;
+        bool return_sequence = options->returnSequence;
+        ngraph_node_t* gruNode = nullptr;
+        status = ngraph_gru_sequence(inputTransposeNode, initialHiddenStateTransposeNode, stepsNode,
+                                     weightNode, recurrentWeightNode, biasNode, hiddenSize,
+                                     direction, linear_before_reset, return_sequence, &gruNode);
+        DAWN_TRY(CheckStatusCode(status, "ngraph gru"));
+
+        std::vector<int64_t> order4D = std::vector<int64_t>{2, 0, 1, 3};
+        const ngraph_node_t* order4DNode =
+            AddConstantWithGraph<int64_t>(precision_e::I64, {order4D.size()}, order4D);
+
+        ngraph_node_t* gruTransposeNode = nullptr;
+        status = return_sequence ? ngraph_transpose(gruNode, order4DNode, &gruTransposeNode)
```

Comment:
How about naming order as order3D... And add some annotations here to describe the transpose purpose, for example, " Transpose it from shape {?} to shape{?} because ...." .

Reply:
Explained in detail.